zkevm-circuits/src/mpt_circuit/mod_extension.rs

```diff
@@ -0,0 +1,327 @@
+        param::HASH_WIDTH,
+        witness_row::StorageRowType,
+        FixedTableTag, RlpItemType,
+    },
+};
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct ModExtensionGadget<F> {
+    rlp_key: [ListKeyGadget<F>; 2],
+    is_not_hashed: [LtGadget<F, 2>; 2],
+    is_short_branch: IsZeroGadget<F>,
+    is_key_part_odd: [Cell<F>; 2], // Whether the number of nibbles is odd or not.
+    mult_key: Cell<F>,
+}
+
+impl<F: Field> ModExtensionGadget<F> {
+    pub fn configure(
+        meta: &mut VirtualCells<'_, F>,
+        cb: &mut MPTConstraintBuilder<F>,
+        ctx: MPTContext<F>,
+        parent_data: &mut [ParentData<F>; 2],
+        key_data: &mut [KeyData<F>; 2],
+    ) -> Self {
+        let mut config = ModExtensionGadget::default();
+
+        circuit!([meta, cb], {
+            let key_items = [
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::LongExtNodeKey as usize,
+                    RlpItemType::Key,
+                ),
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::ShortExtNodeKey as usize,
+                    RlpItemType::Key,
+                ),
+            ];
+            let key_nibbles = [
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::LongExtNodeNibbles as usize,
+                    RlpItemType::Nibbles,
+                ),
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::ShortExtNodeNibbles as usize,
+                    RlpItemType::Nibbles,
+                ),
+            ];
+            let rlp_value = [
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::LongExtNodeValue as usize,
+                    RlpItemType::Value,
+                ),
+                ctx.rlp_item(
+                    meta,
+                    cb,
+                    StorageRowType::ShortExtNodeValue as usize,
+                    RlpItemType::Value,
+                ),
+            ];
+
+            let is_insert = parent_data[0].is_placeholder.expr(); // insert or delete
+
+            let lo_s = is_insert.clone() * parent_data[0].hash.lo().expr()
+                + (1.expr() - is_insert.clone()) * parent_data[1].hash.lo().expr();
+            let hi_s = is_insert.clone() * parent_data[0].hash.hi().expr()
+                + (1.expr() - is_insert.clone()) * parent_data[1].hash.hi().expr();
+            let lo_c = is_insert.clone() * parent_data[0].drifted_parent_hash.lo().expr()
+                + (1.expr() - is_insert.clone()) * parent_data[1].drifted_parent_hash.lo().expr();
+            let hi_c = is_insert.clone() * parent_data[0].drifted_parent_hash.hi().expr()
+                + (1.expr() - is_insert.clone()) * parent_data[1].drifted_parent_hash.hi().expr();
```

Comment:
Yes, totally! Just needed to add `.expr()` after each word, otherwise getting:
```
no method named `mul` found for tuple `(circuit_tools::cell_manager::Cell<F>, circuit_tools::cell_manager::Cell<F>, circuit_tools::cell_manager::Cell<F>, circuit_tools::cell_manager::Cell<F>)`
```